### PR TITLE
Rfc remove armsoftfloatlib

### PR DESCRIPTION
**Pull request by ardbiesheuvel**
Maintaining ArmSoftFloatLib is a bit of a hassle, and as it turns out, mostly unnecessary, now that we have an alternative crypto library in the form of MbedTLS.

So tweak the ARM virtual platforms to switch to MbedTLS for all crypto use except TLS (which has a direct dependency on OpenSSL), and remove ArmSoftFloatLib and the associated git submodule entirely.

### CODE REVIEW

**Comment by ardbiesheuvel** [comment]
@os-d @jyao1 @mdkinney @leiflindholm @samimujawar @lgao4 @liyi77

File: CryptoPkg/Library/OpensslLib/OpensslLib.inf [hunk 1]
```diff
@@ -666,9 +666,6 @@
   DebugLib
   RngLib
 
-[LibraryClasses.ARM]
```
Comment (os-d) [inline CryptoPkg/Library/OpensslLib/OpensslLib.inf:669]: nit: Should we also remove ARM from the list of supported archs for each of these infs? Or leave it since parts of OpenSSL can be used on ARM?

**Review by os-d — APPROVED** [review]
LGTM. If this RFC is approved, I will drop my moving ArmSoftFloatLib to MdePkg.

File: CryptoPkg/Library/OpensslLib/OpensslLib.inf (hunk 1 above)
Comment (ardbiesheuvel) [reply CryptoPkg/Library/OpensslLib/OpensslLib.inf:669]: Yeah, good question. The problem is that OpensslLib is a big pile of related but distinct functionality, and normally (on Linux), it is distributed as separate libcrypto.so and libssl.so shared libraries.

So for ARM, this means that OpensslLib might still be used, as long as the user does not pull in functionality that requires floating point support. Given that MbedTlsLib depends on OpensslLib too (!) -for SM3 support-, we actually still have to provide an OpensslLib resolution on 32-bit ARM too - it will just never reference the objects that require floating point support.

I would prefer for MbedTlsLib not to depend on OpensslLlib at all - the SM3 support case may be important, and I can see how hacking up support this way is acceptable at the product level, I strongly feel that the upstream EDK2 should be held to higher standards in this regard. If MbedTls does not support SM3, and SM3 is a requirement, just use OpensslLib and be done with it.

So TL;DR, I think we should remove this once we drop the OpensslLib dependence from MbedTlsLib.

**Comment by leiflindholm** [comment]
I'm all for this change.

**Comment by os-d** [comment]
@ardbiesheuvel , now that https://github.com/tianocore/edk2/pull/6190 is merged, are we ready for this RFC to be published?
